packages/material-ui-lab/src/TimelineItem/TimelineItem.js

```diff
@@ -1,53 +1,81 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
+import { deepmerge } from '@material-ui/utils';
 import { capitalize, isMuiElement } from '@material-ui/core/utils';
-import { withStyles } from '@material-ui/core/styles';
+import {
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+} from '@material-ui/core/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import TimelineContext from '../Timeline/TimelineContext';
 import TimelineItemContext from './TimelineItemContext';
+import timelineItemClasses, { getTimelineItemUtilityClass } from './timelineItemClasses';
 
-export const styles = () => ({
-  /* Styles applied to the root element. */
-  root: {
-    listStyle: 'none',
-    display: 'flex',
-    position: 'relative',
-    minHeight: 70,
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(
+    {
+      ...styles[`align${capitalize(styleProps.align)}`],
+      [`& .${timelineItemClasses.content}`]: styles.content,
+      [`& .${timelineItemClasses.oppositeContent}`]: styles.oppositeContent,
+    },
+    styles.root || {},
+  );
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { align, classes } = styleProps;
+
+  const slots = {
+    root: ['root', `align${capitalize(align)}`],
+    content: ['content'],
+    oppositeContent: ['oppositeContent'],
+  };
+
+  return composeClasses(slots, getTimelineItemUtilityClass, classes);
+};
+
+const TimelineItemRoot = experimentalStyled(
+  'li',
+  {},
+  {
+    name: 'MuiTimelineItem',
+    slot: 'Root',
+    overridesResolver,
   },
-  /* Styles applied to the root element if `align="left"`. */
-  alignLeft: {},
-  /* Styles applied to the root element if `align="right"`. */
-  alignRight: {
+)(({ styleProps }) => ({
+  listStyle: 'none',
+  display: 'flex',
+  position: 'relative',
+  minHeight: 70,
+  ...(styleProps.align === 'right' && {
     flexDirection: 'row-reverse',
-  },
-  /* Styles applied to the root element if `align="alternate"`. */
-  alignAlternate: {
+  }),
+  ...(styleProps.align === 'alternate' && {
     '&:nth-child(even)': {
       flexDirection: 'row-reverse',
-      '& $content': {
+      '& .MuiTimelineItem-content': {
```

Comment:
```suggestion
      '& .MuiTimelineContent-root': {
```

Reply:
Shouldn't we import the JS classes instead of hard coding them?

Reply:
Yeah, even better, my point was that we don't need to use context for the classes. Let me update and merge this PR 👍